lib/libc/netdb/lib_getaddrinfo.c

```diff
@@ -0,0 +1,98 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+
+#include <netdb.h>
+
+/****************************************************************************
+ * Private Data Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: getaddrinfo
+ *
+ * Description:
+ * Translates the name of a service location (for example, a host name) and/or
+ * a service name and returns a set of socket addresses and associated
+ * information to be used in creating a socket with which to address the
+ * specified service.
+ *
+ * Due to a limitation in dns_gethostbyname, only the first address of a
+ * host is returned.
+ * Also, service names are not supported (only port numbers)!
+ *
+ * Input Parameters:
+ *   hostname  - descriptive name or address string of the host
+ *                 (may be NULL -> local address)
+ *   servname - port number as string of NULL
+ *   hint - structure containing input values that set socktype and protocol
+ *   res - pointer to a pointer where to store the result (set to NULL on failure)
+ *
+ * Returned Value:
+ *   0 on success, -1 on failure of ioctl(),
+ *   one of below error codes on failure of lwip_getaddrinfo()
+ *   EAI_NONAME      200
+ *   EAI_SERVICE     201
+ *   EAI_FAIL        202
+ *   EAI_MEMORY      203
+ *   EAI_FAMILY      204
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: getaddrinfo
+ ****************************************************************************/
+#ifdef CONFIG_NET_LWIP_NETDB
+int getaddrinfo(FAR const char *hostname, FAR const char *servname, FAR const struct addrinfo *hint, FAR struct addrinfo **res)
+{
+	int ret = -1;
+	struct lwip_info info_s;
+
+	int sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sockfd < 0) {
+		return ret;
```

Comment:
This is an error case, why don't you put debug message & print out errno?

Reply:
ok